model-datastructure/src/commonMain/kotlin/org/modelix/model/VersionMerger.kt

```diff
@@ -116,6 +117,24 @@ class VersionMerger(private val storeCache: IDeserializingKeyValueStore, private
         return mergedVersion!!
     }
 
+    /**
+     * Instead of computing and applying the inverse operation we can optimize for the case when an undo follows
+     * directly after the version to be undone and just drop the version.
+     */
+    private fun filterUndo(versions: List<CLVersion>): List<CLVersion> {
```

Comment:
Isn't there a chance for a more readable version of this method by using [`windowed`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/windowed.html).